policy/modules/admin/fapolicyd.fc

```diff
@@ -0,0 +1,13 @@
+/etc/fapolicyd(/.*)?				gen_context(system_u:object_r:fapolicyd_config_t,s0)
+/etc/fapolicyd/compiled\.rules	--	gen_context(system_u:object_r:fapolicyd_compiled_rules_t,s0)
```

Comment:
I setup a different type for compiled.rules because it is compiled by fagenpolicy (which writes as fapolicyd_conf_t but then calls restorecon at the end of the script).  This way the compiled rules are read-only to everyone.
Alternately fagenpoilcy could write to fapolicyd_compiled_rules_t and the restorecon becomes redundant.

Reply:
Looks to be too many issues with MLS (beyond fapolicy) will look at that in the future.

Reply:
>Alternately fagenpoilcy could write to fapolicyd_compiled_rules_t and the restorecon becomes redundant.

If that's the only use of /tmp, that can work. If not, then we stick with what you've already written. Please add a comment in the policy explaining this.
 
> Though, I just noticed the comment before the restorecon:
> 
> ```
> # Restore context on MLS system.
> # /tmp is SystemLow & fapolicyd.rules is SystemHigh
> ```
> 
> So, maybe the context should be "gen_context(system_u:object_r:fapolicyd_compiled_rules_t,mls_systemhigh)"

It doesn't seem like this daemon would be usable on an MLS system. If the file is supposed to be world-readable, then that wouldn't work because it could expose paths/filenames that a process doesn't have access to due to its current level.

Reply:
I wavered back and forth here and gave fagenrules_t write access to fapolicyd_compiled_rules_t as it is copying those files into the directory (with a type transition).  I think that is clearer than how it was (though I may have forgotten to edit comments - I will check that).